### PR TITLE
Update Rack to support version 300

### PR DESCRIPTION
**Pull request by abiliogp**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve. e.g., Fixes #01]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: lib/puppet/provider/oneview_rack/c7000.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ################################################################################
-# (C) Copyright 2016-2017 Hewlett Packard Enterprise Development LP
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
```
Comment (fgbulsoni) [inline lib/puppet/provider/oneview_rack/c7000.rb]: No need to alter this

File: lib/puppet/provider/oneview_rack/synergy.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+Puppet::Type.type(:oneview_rack).provide :synergy, parent: :c7000 do
+  desc 'Provider for OneView Rack using the Synergy variant of the OneView API'
+
+  confine true: login[:hardware_variant] == 'Synergy'
+
+  def initialize(*args)
```
Comment (fgbulsoni) [inline lib/puppet/provider/oneview_rack/synergy.rb]: Should not be needed anymore

File: lib/puppet/provider/oneview_rack/c7000.rb [hunk 3]
```diff
@@ -14,33 +14,18 @@
...
+
+Puppet::Type::Oneview_rack.provide :c7000, parent: Puppet::OneviewResource do
+  desc 'Provider for OneView Rack using the C7000 variant of the OneView API'
+
+  confine true: login[:hardware_variant] == 'C7000'
 
-Puppet::Type.type(:oneview_rack).provide(:oneview_rack) do
   mk_resource_methods
 
   def initialize(*args)
```
Comment (fgbulsoni) [inline lib/puppet/provider/oneview_rack/c7000.rb]: Should not be needed anymore

**Review by fgbulsoni — CHANGES_REQUESTED** [review]
- Improve description and list the issues addressed. 
- Should also update integration tests and if needed, the examples. Make sure the integrations/examples are passing.
- Also verify code coverage

File: lib/puppet/provider/oneview_rack/c7000.rb (hunk 3 above)
Comment (abiliogp) [reply lib/puppet/provider/oneview_rack/c7000.rb]: def initialize(*args)?

File: lib/puppet/provider/oneview_rack/synergy.rb (hunk 2 above)
Comment (abiliogp) [reply lib/puppet/provider/oneview_rack/synergy.rb]: def initialize(*args)?

**Review by fgbulsoni — APPROVED** [review]

**Pull request merged**